File: apps/monolith/src/main/java/com/example/iter/admin/action/dto/response/AdminActionResponse.java
package com.example.iter.admin.action.dto.response;

import com.example.iter.common.audit.domain.entity.AdminActionTargetType;
import com.example.iter.common.audit.domain.entity.AdminActionType;

import java.time.LocalDateTime;

public record AdminActionResponse(
        Long actionId,
        Long adminId,
        AdminActionTargetType targetType,
        Long targetId,
        AdminActionType action,
        String reason,
        LocalDateTime createdAt
) {
}

File: apps/monolith/src/main/kotlin/iter/admin/action/dto/response/AdminActionResponse.kt
package iter.admin.action.dto.response

import iter.common.audit.domain.entity.AdminActionTargetType
import iter.common.audit.domain.entity.AdminActionType
import java.time.LocalDateTime
import kotlin.jvm.JvmRecord

@JvmRecord
data class AdminActionResponse(
    val actionId: Long?,
    val adminId: Long,
    val targetType: AdminActionTargetType,
    val targetId: Long,
    val action: AdminActionType,
    val reason: String?,
    val createdAt: LocalDateTime?
)
